articles/variables-in-js/index.md

```diff
@@ -0,0 +1,129 @@
+# Understanding Variables In Javascript
+
+## Introduction
+Variables are a way for programmers to name a value to reuse it, update it, or keep track of it.
+In this article, we will go through naming variables, declaring them, and initializing them.
+We are also going to touch on the scope of variables to ensure that we understand variables fully.
+
+## Prerequisites
+To follow along with this tutorial, you need a web browser. You can get Google Chrome [here](https://www.google.com/chrome/)
+
+### Accessing the JavaScript Console
+We will use the JavaScript console to execute the JavaScript code in this tutorial. To open it in Google Chrome, press `Ctrl + Shift + J` on Windows and Linux and `Cmd + Option + J` in Mac OS.
+
+### Understanding the scope of variables
+Scope determines the accessibility of variables, objects, and functions from different parts of your code.
+Javascript has two types of scopes, which are:
+1. Global scope
+2. Local scope
+
+**1. Global scope**
+
+When a variable is *globally scoped*, it means it is available anywhere in your program. 
+Declaring a variable outside a function leads to it being ade to the global scope automatically.
+Global scope uses `var` key word to declare variables.
+
+**2. Local scope**
+
+When variables are declared within a function they become locally scoped, that means they can only be called inside the function.
+Local scope variables are divided into:
+- function scoped variables
+- block scope variables
+
+**a.Function scoped variables**
+
+When you say a variable is function scoped, it means that the variable defined within the function is not accessible from outside the function.
+To define a locally scoped variable we use the `var` keyword.
+
+
+**b.Block scoped variables**
+
+In JavaScript a block is denoted by curly braces, the space between the curly brackets is what is known as a block.
+For example: the `if...else, do...while` and `for` loop statements create blocks.
+
+When a variable is block-scoped, it means that it exists inside the block, it was defined.
+
+`let` and `const` keywords are used in defining the block scope.
+
+### Naming Variables In JavaScript
+Before you start declaring variables, you should first learn how to name them first. Below are some of the rules that one should follow to ensure that your JavaScript code is easy to understand and work with. Variable names;
+- should not have spaces.
+- should begin with an underscore, a dollar sign or a letter.
+- are very case-sensitive. This means `car_TYPE` is treated as an entirely different variable than one named `car_type`
+- must contain only letters, numbers, underscores, or dollar signs.
+- You can't use any JavaScript *reserved words* as a variable name, reserved words include; private, public, enum etc.
```

Comment:
These rules are not to make the code easily readable. Failing to follow these rules will result in an error.

Reply:
Capitalize the first letters in each point.